src/app/weather-forecast/weather-forecast.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, Input, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { WeatherFeature } from '../model';
+import { WeatherForecastApiService } from './weatherForecastApiService/weather-forecast-api.service';
+
+@Component({
+    selector: 'app-weather-forecast',
+    templateUrl: './weather-forecast.component.html',
+    styleUrls: ['./weather-forecast.component.scss'],
+})
+
+export class WeatherForecastComponent implements OnDestroy {
+    @Input() city = 'Pisa';
+    @Input() state = 'it';
+    weather: WeatherFeature;
+    subscription: Subscription;
+
+    constructor(public weatherData: WeatherForecastApiService) {
+
+        this.subscription = this.weatherData.getWeatherData(this.city, this.state).subscribe((response) => {
+            const ks: string[] = Object.keys(response);
+            const subKeys: string[] = ['temp', 'pressure'];
```

Comment:
The right way to do this would be defining a type for what the api returns. But for now I will accept this.